crates/agent/src/discovers/specs.rs

```diff
@@ -0,0 +1,373 @@
+use std::collections::BTreeMap;
+
+#[derive(serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DiscoveredBinding {
+    /// A recommended display name for this discovered binding.
+    pub recommended_name: String,
+    /// JSON-encoded object which specifies the endpoint resource to be captured.
+    pub resource_spec: models::Object,
+    /// JSON schema of documents produced by this binding.
+    pub document_schema: models::Schema,
+    /// Composite key of documents (if known), as JSON-Pointers.
+    #[serde(default)]
+    pub key_ptrs: Vec<models::JsonPointer>,
+}
+
+pub fn parse_response(
+    endpoint_config: &serde_json::value::RawValue,
+    image_name: &str,
+    image_tag: &str,
+    response: &[u8],
+) -> Result<(models::CaptureEndpoint, Vec<DiscoveredBinding>), serde_json::Error> {
+    let image_composed = format!("{image_name}{image_tag}");
+    tracing::debug!(%image_composed, response=%String::from_utf8_lossy(response), "converting response");
+
+    // Response is the expected shape of a discover response.
+    #[derive(serde::Deserialize)]
+    struct Response {
+        #[serde(default)]
+        bindings: Vec<DiscoveredBinding>,
+    }
+    let Response { mut bindings } = serde_json::from_slice(response)?;
+
+    // Sort bindings so they're consistently ordered on their recommended name.
+    // This reduces potential churn if an established capture is refreshed.
+    bindings.sort_by(|l, r| l.recommended_name.cmp(&r.recommended_name));
+
+    Ok((
+        models::CaptureEndpoint::Connector(models::ConnectorConfig {
+            image: image_composed,
+            config: endpoint_config.to_owned(),
+        }),
+        bindings,
+    ))
+}
+
+pub fn merge_capture(
+    capture_name: &str,
+    endpoint: models::CaptureEndpoint,
+    discovered_bindings: Vec<DiscoveredBinding>,
+    fetched_capture: Option<models::CaptureDef>,
+    update_only: bool,
+) -> (models::CaptureDef, Vec<DiscoveredBinding>) {
+    let capture_prefix = capture_name.rsplit_once("/").unwrap().0;
```

Comment:
This is the most-specific location of the changed behavior: we used to use the capture_name attached to the discover operation as the _entire_ prefix, to which we would suffix the trailing component of the connector name to form the capture name. Now we take it as the literal capture name, and used its base name as the `capture_prefix`.

Compare to the prior behavior of swizzle_response_to_specs (sp ?)